### ISSUE TITLE
running from everything: is it optimal with the new experience system?

### ISSUE THREAD

**Issue by jeremygurr** [jeremygurr/dcssca#133]
I tried a couple of spriggans with the new experience mechanic, in an attempt to avoid almost all fights and see if I could win. I didn't even get close of course, but I noticed some interesting things. 

First, I don't think the new experience mechanic means that optimal play requires avoiding all monsters. The easiest way to do this is just to dive as deep as possible as quickly as possible, but then you miss most of the experience potions, which is where you will get most of your character growth. Spending much time hunting for the potions on a level inevitably puts you into narrow passageways with monsters on both sides, making it very difficult to pull off. And even if you did manage to gather up all of the experience potions on the floor, you will still be missing a lot of experience gained from potions dropped by uniques. This means that you will be grossly underpowered when it comes to obtaining actual runes or the orb. This aspect is not much different than mainstream crawl, where if you minimize monster kills, you end up underpowered at the end. 

This being said, I think a new play style is made possible by this change. It is a super difficult challenge mode, similar to speed running, but even more difficult, since you are avoiding all conflict, and trying to gather as much experience potions as you can at the same time. There's a lot of strategy potential here, like avoiding narrow passageways, focussing purely on evasion and escape magic, etc. Even though I was destroyed each time I tried this, I still had fun in the process. I didn't find it tedious, since my actions were both clearly motivated, meaningful, and very challenging. I think with enough skill, a player could pull off a pacifist win, which would be very impressive.

**Comment by Sandman25DCSS** [comment]
Do ghosts/uniques give the same potions of experience as generated ones? I suspect some games can be extremely lucky on uniques (several per floor) so the game could become too easy. I don't think standard crawl has uniques (except TRJ probably) who give as much XP as the rest of the floor they are met on.
Even if peaceful runs are impossible, Spriggan still can explore everything without killing most monsters (at least with old_movement=true).

**Comment by Sandman25DCSS** [comment]
Have you tried SpAs of Dith who puts all experience into Stealth and later into Invocations for Shadow Form?

**Comment by Sandman25DCSS** [comment]
My point is that it is not optimal to run from everything but it is optimal to run from many dangerous monsters. For example, Spriggan in Snake will rarely want to kill Greater Nagas or Naga Warriors.

**Comment by jeremygurr** [comment]
the potion of experience for ghosts and uniques is the same as the one that spawns on the floor. Keep in mind that the amount of experience gained from drinking a potion of experience on a floor is substantially less than the old crawl would give for killing all of the creatures on the floor. For example, in the old way I would often be at level 4 or even 5 by D:2. With the new way, I'm usually at level 1 or 2 even as deep as D:4, although I'm betting against being able to go deeper. But even after plunging down to say D:8, and then drinking all of my experience potions, I still would be lucky to be around level 8, whereas before I'm usually past level 10 by then. So on average I think it's a little less than the experience given before, but it provides the option of taking greater risks and getting more experience than before (although I haven't actually pulled that off yet, usually I still fall short of where I would normally be). 

I could easily take away the potions of experience from uniques and ghosts. However, the guaranteed drop is a big incentive to take larger risks, which makes the game more exciting. Also, instead of seeing a powerful ghost or unique and thinking "this sucks", the player thinks "cool!". Even if the player can't tackle a unique right away, they have an incentive to come back and finish them off, which I think is fun. Plus it gives them something to do as they become more powerful, since experience potions never diminish in value, no matter how powerful or advanced the player is. 

So if the problem is that there is still too much experience in the game, it is easy to cut that back a little. I think that is better than taking exp potions from uniques and ghosts, and taking away that element of excitement.

**Comment by jeremygurr** [comment]
SpAs of Dith sounds fun. I do want to try more "runner" games. These I think have a different feel than speed runs, where the objective is to minimize turns. In a runner game, you can take your time, but you still have the excitement of having to deal with vastly overpowered monsters that you have in a speed run. But you have to more heavily invest in escape strategies than a normal game would require.

**Comment by jeremygurr** [comment]
Isn't is always optimal to run away from dangerous monsters? How is that different than before? The incentive to kill every single monster is taken away, that is true, but you still have the incentive to take out monsters that are between you and your objective, or that you can't escape from. 

Removing the incentive to kill every single monster I think actually cuts back on some of the less exciting parts of the game. It does cut out a little excitement when it comes to challenging some hard non-uniques, but that is replaced by much more excitement added because of the challenge caused by hoarding experience potions. Overall I think you end up with less of the boring stuff and more of the exciting stuff.

**Comment by Sandman25DCSS** [comment]
Standard crawl encourages killing monsters for XP but still makes it possible to avoid the most dangerous monsters (no matter unique, ghost or just Juggernaut/Caustic Shrike/Iron Giant/OoF).
Standard crawl encourages killing "killable" monsters as soon as possible since the amount of XP does not depend on your level.
Your mod encourages avoiding all monsters, especially uniques/ghosts because
1) you are advocating to suspend quaffing potions of experience for as long as possible. It does not make much sense to try killing a unique/ghost if you are not going to quaff the potion immediately. By the way it reminds me gambling somewhat, I suspect there can be many deaths with multiple potions of experience. I suggest to change potion of experience to give you the same XP no matter where you are. Otherwise we can have weird situation when all "easy early game" species automatically become "easy late game" species because they quaff those potions really late, and all "hard early game" species automatically become "hard late game" species because they quaff all XP potions as soon as possible just to survive.
2) after finding potion from D:1 it is optimal to dive as deep as possible to quaff your first potion of experience (D:10, for example, if you had to retreat as your first action after entering D:11), then return to D2 to fully explore it, find corresponding potion/uniques, return to D:10 again to quaff the potion(s), return to D:3 to fully explore it, find corresponding potion/uniques, return to D:10 again to quaff the potion and so on. That's time-consuming and IMHO boring. It reminds me grinding/farming, getting full XP reward for killing an easy unique/ghost/exploring level is not what I like.
Basically I don't see why you are combining incentive to avoid monsters (even extremely dangerous monsters don't give any XP, right?) with incentive to kill harmless monsters to get a huge award as long as they are uniques/ghosts. Personally I would be more happy with "no XP for harmless/weak monsters, some minor XP for dangerous, huge XP for extremely dangerous" no matter if they are uniques/ghosts or just Water Moccasin on D:1.

**Comment by Sandman25DCSS** [comment]
In standard crawl it is not always optimal to run away from dangerous monsters, I often choose to kill them with some consumable used because it will give me some XP (and eventually HP/MP), it is often better to have less consumables but more powerful character, it results in higher chance to win the game.
In your "no XP for monsters" mod using consumable to kill a dangerous monster is a forced decision which happens only when player cannot escape, it results in just lost consumable and the same character, chance to win is decreased.

**Comment by Sandman25DCSS** [comment]
Of course giving more XP for extremely dangerous monsters is quite the opposite of what you are trying to achieve, it would make life of "runners" and "limited pacifists" much harder than in standard crawl.

**Comment by jeremygurr** [comment]
standard crawl also encourages avoiding uniques if you can't safely kill them, since they will give you just as much experience later, when the player is stronger and has an easier time with it. 

As far as whether a species is easy early on or later, that is something that is much more in the player's control now than it has been in the past. In the past, they are pretty much stuck with a fixed experience structure, and it is easy or hard depending on factors that they don't have a lot of control over. With this change, they can make early game easier, or late game easier than it would otherwise be by deliberate choices and risks that are taken. This does increase the learning curve for a new player, and would require a good enhancement to the tutorial to really address that properly, but I'm not against making the game require more learning if it is easy to grasp, intuitive strategies, I'm just against a large number of obscure rules that a player has to look up and memorize to really be able to play well.

Yes it is optimal to save your potion for as long as possible, but that doesn't necessarily mean diving immediately is optimal. I've played multiple strategies several times now and I'm not sure that one is absolutely superior to another. For example, you can grab your first potion, then do a power dive, gathering up a lot of experience just by reaching new floors, and then go back up. But a second powerdive would not be nearly as rewarding, since you've already visited those levels, so there is more incentive to gather a few more potions and consumables first. Another equally viable option is to just clear each floor carefully, saving the potions until you reach the floor you would normally reach in your power dive. In this second scenario, the player gets just as much experience as the first, but faces risk in a different way because of whatever encounters have to be dealt with along the way. The optimal strategy here is probably something in between the two, drinking an occasional exp potion at a non-optimal point to keep the player alive. Deciding exactly when to do it isn't a trivial problem, and would take a lot of game experience to know the right times and effective strategies, which can change for different species. 

If this resulted in reducing player options and increasing tedium, then I would agree. But so far I'm not seeing that. Each time I play I come up with new strategies to try to see if I can push the limits a little further, and it becomes more exciting than ever before. Or I can easily play a more relaxed game by drinking the experience potions a little sooner. 

From what I've seen so far, it is not optimal to dive and return to the surface repeatedly. The optimal path shifts according to what I come across in each level, and what risks I think I can pull of. It does mean criss-crossing the dungeon more than before, but that hasn't been boring yet, because inevitably I'm not clearing everything in between, so travel each way is always eventful, and often quite rewarding as additional items or potions are found. Even just returning to the surface is very dangerous, since even after the first dive, you are still quite underpowered to deal with what is lurking between. 

There is still plenty of incentive to take out dangerous non-uniques, as they make travel quite hazardous otherwise. This becomes painfully apparent when attempting pacifist runs. They also have a better chance of dropping more valuable items. 

The objective here isn't to make the pacifist strategy possible. That's just an interesting side effect. The purpose here is to eliminate experience grinding (forcing the player to kill many easy, non-interesting monsters to maximize their experience) and increase the excitement and choices for the player. Allowing the players more options about what to avoid and what to engage I think is part of achieving this. I don't think it's a bad thing for some monsters to be avoided entirely. That certainly happens in standard crawl. 

The monsters that are a more serious threat to the player are also the ones a player has more incentive to kill, precisely because of the threat level. This gives me an interesting idea though... what if new monster spawns on a level have a higher chance of spawning creatures that already exist on the level (after the initial population of the level of course)? That would make levels automatically adapt to create monsters the player has more trouble with. If I left a few death yaks on a level, I may come back later to find a large herd of them. This could be particularly important to consider for the orb ascension strategy. 

I'm not seeing the need to add further incentive to dangerous monsters. So what if I avoid the occasional hydra or death yak? Although it's definitely not as easy as it used to be with the new movement rules, again adding to the incentive to take care of them sooner than later if possible. I think the small loss of excitement from having to deal with the tougher monsters is more than made up for by the added excitement of the game as a whole getting a lot more difficult because I want to push the limits to maximize my experience. 

This effectively becomes an auto-difficulty system that better matches difficulty level to the player than any game mechanic I've ever seen. Even playing on easy mode, my games get insanely hard just minutes after starting. And if the incentive to go back and kill boring easy uniques to get added experience becomes a common source of tediousness in the game, I can easily remove it, and still preserve the thrill that currently exists. 

Would you like me to add an option to disable unique/ghost exp potion drops so it can be tried out? It would be fairly easy. I think it would effectively address one of your concerns, although I don't see the need for it yet, because I tend to go back and clear out the uniques as quickly as I'm strong enough to do it, so it's never a situation of being vastly overpowered and having to engage in a few dozen boring battles later in the game. 

An even more extreme option would be to eliminate experience potions entirely, and only award experience for reaching new floors. Then the only incentive for returning to higher floors is to get more consumables / better loot. Which is still a pretty good reason, but not quite as compelling. In this situation you may still want to dive multiple times as you get strong enough to go further.

**Comment by jeremygurr** [comment]
The more I think of it, the more it seems to be a good idea to get rid of all monster potion drops. That simplifies things from a player's perspective. They still have an incentive to kill monsters for items and to reduce danger. And more powerful monsters ought to have better chances of dropping good stuff (which I think is already the case), increasing the incentive to take out more powerful monsters. 

Maybe I will do that, and we can try it that way.

**Comment by jeremygurr** [comment]
And yes, I've died many times with lots of potions of experience. Although in standard crawl I often embarrassingly end up dead with potions of healing or blink scrolls, which IMO is even worse.

**Comment by Sandman25DCSS** [comment]
Ok, let me try the mod, maybe my concerns are based on invalid assumptions.
No, I wouldn't like option for disabling potions for uniques/ghosts, it is close to introducing a new difficulty mode "impossible" ;)

Dying with blink scrolls/potions of healing is almost always a serious player mistake, dying with potions of exp is probably not.

**Comment by Sandman25DCSS** [comment]
This is what I meant.
http://crawl.homedns.org/morgue/Sandman25/Sandman25.txt

Health: 10/10      AC:  4    Str:  7    XL:     2   Next: 66%
Magic:  7/7        EV: 17    Int: 12    God:    Dithmenos [......]
Gold:   91         SH:  0    Dex: 19    Spells: 0 memorised, 1 level left

  Skills:
- Level 1.1 Fighting
- Level 2.1 Short Blades
- Level 2.0 Throwing
- Level 1.7 Dodging
- Level 9.2 Stealth

c - 5 potions of experience {!q}

You are on level 6 of the Dungeon.
Vanquished Creatures
  An adder (D:4)
1 creature vanquished.

I had to kill the adder because it cornered me into a deep end corridor.

**Comment by Sandman25DCSS** [comment]
It is my first attempt

**Comment by jeremygurr** [comment]
I'm not sure what this means. You seem to be showing me that you didn't have to kill anything in the first 6 levels?

**Comment by Sandman25DCSS** [comment]
Yes, I am showing you that it is possible to start-scum to get very easy game due to luck in early levels.
Or just use easy combos and don't depend on luck.
So gap between easy combos and hard combos increases.

**Comment by jeremygurr** [comment]
Oh you are talking about how you collected 5 exp potions without killing anything? I still don't think that changes your overall game that much. You are missing all of the consumables that you have an easy time getting in those first few levels. Unless you are specifically trying for a pacifist run, the easier long term strategy is to clear the first few easy levels before diving deep to gather a bunch of experience. 

I think what you actually gain by this particular kind of scumming is too little for it to have any real attraction. 

Now for the species that have a hard time early game, they may not be able to immediately take the kind of risks that an easier species could, but as they get a bit stronger, they can still handle some good amount of risk, and the benefit of saving exp potions is the same whether done early game or late game.

**Comment by Sandman25DCSS** [comment]
I will try 50% XP from monsters and 50% XP from potions and will see how it goes.
